From ac050e0b4e8e8cfe1cefb83a499e87ab9f9edfdd Mon Sep 17 00:00:00 2001
From: John Cox <jc@kynesim.co.uk>
Date: Thu, 1 Oct 2026 19:26:25 +0100
Subject: [PATCH 26/40] qt: Improve wayland platform detection

Wayland platform string can be wayland-* so match prefix rather
than entire string
---
 modules/gui/qt/qt.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/modules/gui/qt/qt.cpp
+++ b/modules/gui/qt/qt.cpp
@@ -667,7 +667,8 @@ static void *ThreadPlatform( void *obj,
         QString platform = app.platformName();
         if( platform == qfu("xcb") )
             p_sys->voutWindowType = VOUT_WINDOW_TYPE_XID;
-        else if( platform == qfu("wayland") )
+        /* Qt reports "wayland", "wayland-egl" or "wayland-xcomposite-*/
+        else if( platform.startsWith( qfu("wayland") ) )
             p_sys->voutWindowType = VOUT_WINDOW_TYPE_WAYLAND;
         else if( platform == qfu("windows") )
             p_sys->voutWindowType = VOUT_WINDOW_TYPE_HWND;
